common/unified/matrix/dense_kernels.cpp

```diff
@@ -275,6 +273,23 @@ void compute_norm2(std::shared_ptr<const DefaultExecutor> exec,
 GKO_INSTANTIATE_FOR_EACH_VALUE_TYPE(GKO_DECLARE_DENSE_COMPUTE_NORM2_KERNEL);
 
 
+template <typename ValueType>
+void count_nonzeros_per_row(std::shared_ptr<const DefaultExecutor> exec,
+                            const matrix::Dense<ValueType>* mtx,
+                            size_type* result)
+{
+    run_kernel_row_reduction(
+        exec,
+        [] GKO_KERNEL(auto i, auto j, auto mtx) {
+            return mtx(i, j) != zero(mtx(i, j)) ? 1 : 0;
```

Comment:
Maybe change it to `is_zero` instead ?

Reply:
doesn't look like we have such a function right now?

Reply:
can we easily add it ?

Reply:
sounds like a good idea. probably makes a lot of conversion code simpler/more literal